nsqd/nsqd.go

```diff
@@ -515,6 +516,10 @@ func (n *NSQD) GetTopic(topicName string) *Topic {
 	t = NewTopic(topicName, &context{n}, deleteCallback)
 	n.topicMap[topicName] = t
 
+	if atomic.LoadInt32(&n.isLoading) == 0 {
+		t.UnPause()
```

Comment:
I wonder if we can move this below the channels block, as an alternative to taking the topic lock to block PutMessages() while getting channels.

Reply:
The `channelMap` in `getOrCreateChannel` needs the topic's lock . I am not sure if we can do this~

```go
// this expects the caller to handle locking
func (t *Topic) getOrCreateChannel(channelName string) (*Channel, bool) {
    channel, ok := t.channelMap[channelName]
```

Reply:
Right - you'd call `t.GetChannel()` instead. But this is still tentative supposition :)